### PR TITLE
requirements: update jsonschema dependency

### LINKED ISSUE #6347: Update jsonschema dependency
**Is your feature request related to a problem? Please describe.**
I'm trying to install [poetry](https://github.com/sdispater/poetry) and `docker-compose` in a virtualenv but since they depend on different [jsonschema](https://github.com/Julian/jsonschema/blob/master/CHANGELOG.rst) major versions they become incompatible when running together in the venv.

- `poetry`: [`>=3, <4`](https://github.com/sdispater/poetry/blob/master/pyproject.toml#L30)
- `docker-compose`: [`>=2.5.1, <3`](https://github.com/docker/compose/blob/master/setup.py#L42)

**Describe the solution you'd like**
`docker-compose` to update its `jsonschema` dependency to `>=3, <4`.

**Describe alternatives you've considered**
I could just put the two in different venvs. I'm lazy and just want all my tools available in a single venv so I don't have to hop around so much.

### PR DESCRIPTION
**Pull request by chibby0ne**
Fixes: https://github.com/docker/compose/issues/6347

Signed-off-by: Antonio Gutierrez <chibby0ne@gmail.com>

Resolves #6347

### CODE REVIEW

**Comment by chibby0ne** [comment]
Looks like PyInstaller is failing for jsonschema 3.0.1, will take a look at this a bit more.

**Comment by chibby0ne** [comment]
I've made a PR in pyinstaller that should fix the build issue: https://github.com/pyinstaller/pyinstaller/pull/4168, after that is merged and released, I'll update the pyinstaller dependency and this PR should pass the CI.

**Comment by ijc** [comment]
Hi @chibby0ne looks like you https://github.com/pyinstaller/pyinstaller/pull/4168 is merged. Will you have time to rebase?

**Comment by chibby0ne** [comment]
Hi @ijc I've rebased and also pinned the version of `pyinstaller` to the commit containing my PR, since there hasn't been a 3.5 release yet.

**Comment by chibby0ne** [comment]
OK fixed the failing build. Took me a while longer since some things changed in the mean time on the `linux-entrypoint` so had to change that too.

**Comment by chibby0ne** [comment]
@ijc friendly ping that this is ready to be merged

**Comment by OrangeTux** [comment]
What is the status of this?

**Comment by chibby0ne** [comment]
> What is the status of this?

I've resolved the conflicts that appeared in the meantime while waiting for this PR to get merged. Now I'm just waiting for a committer/maintainer to approve/merge.

File: requirements-build.txt [hunk 1]
```diff
@@ -1 +1 @@
-pyinstaller==3.4
+git+git://github.com/pyinstaller/pyinstaller@4f051b301b0ed0a79b8605a07719222c19459e3d#egg=pyinstaller
```
Comment (ulyssessouza) [inline requirements-build.txt]: What about just bumping `pyinstaller` version to 3.5?

File: script/build/linux-entrypoint [hunk 2]
```diff
@@ -20,10 +20,12 @@ echo "${DOCKER_COMPOSE_GITSHA}" > compose/GITSHA
...
 if [ ! -z "${BUILD_BOOTLOADER}" ]; then
-    # Build bootloader for alpine
-    git clone --single-branch --branch master https://github.com/pyinstaller/pyinstaller.git /tmp/pyinstaller
+    # Build bootloader for alpine; develop is the main branch
+    git clone --single-branch --branch develop https://github.com/pyinstaller/pyinstaller.git /tmp/pyinstaller
     cd /tmp/pyinstaller/bootloader
-    git checkout v3.4
+    # Checkout commit corresponding to version in requirements-build; should update to v3.5 when available
+    # This version should issue: https://github.com/docker/compose/issues/6347
+    git checkout 4f051b301b0ed0a79b8605a07719222c19459e3d
```
Comment (ulyssessouza) [inline script/build/linux-entrypoint]: Same as above. I would say that trying `git checkout v3.5` would be cleaner.

File: setup.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ def find_version(*file_paths):
     'docker[ssh] >= 3.7.0, < 5',
     'dockerpty >= 0.4.1, < 1',
     'six >= 1.3.0, < 2',
-    'jsonschema >= 2.5.1, < 3',
+    'jsonschema >= 3, < 4',
```
Comment (ulyssessouza) [inline setup.py]: Please let the inferior bound as it is, since it still works with versions `>= 2.5.1`

File: requirements.txt [hunk 4]
```diff
@@ -11,14 +11,14 @@ enum34==1.1.6; python_version < '3.4'
...
-jsonschema==2.6.0
+jsonschema==3.0.1
 paramiko==2.4.2
 pypiwin32==219; sys_platform == 'win32' and python_version < '3.6'
 pypiwin32==223; sys_platform == 'win32' and python_version >= '3.6'
 PySocks==1.6.7
 PyYAML==4.2b1
 requests==2.22.0
-six==1.10.0
+six==1.12.0
```
Comment (ulyssessouza) [inline requirements.txt]: Is it necessary to bump `six` to bump `jsonschema` ? If not, please issue that in another issue/PR

File: requirements-build.txt (hunk 1 above)
Comment (chibby0ne) [reply requirements-build.txt]: At the moment I created the PR 3.5 wasn't released yet. It's good to know they finally did another release. I will bump it accordingly.

File: script/build/linux-entrypoint (hunk 2 above)
Comment (chibby0ne) [reply script/build/linux-entrypoint]: ditto

File: setup.py (hunk 3 above)
Comment (chibby0ne) [reply setup.py]: Seems reasonable.

File: requirements.txt (hunk 4 above)
Comment (chibby0ne) [reply requirements.txt]: Well pip is able to install jsonschema >= 3 using six==1.10.0 but shows this error:

```
ERROR: jsonschema 3.0.1 has requirement six>=1.11.0, but you'll have six 1.10.0 which is incompatible
```

I would suggest therefore bumping it.

**Comment by chibby0ne** [comment]
> Hello @chibby0ne ! Thank you for the contribution!
> I just had some comments on some version pinning

Thanks for the review @ulyssessouza. Made the suggested changes and rebased to latest master.

**Comment by chibby0ne** [comment]
Hi all friendly ping that this PR is ready to be merged :)!

**Comment by graingert** [comment]
@rumpl can you give this a review please?

**Comment by rumpl** [comment]
Hey @graingert, we will release a new rc as soon as #6584 is merged. Shouldn't take that long, the PR is almost ready to be merged.

**Comment by graingert** [comment]
looks like it got moved to https://github.com/docker/compose/pull/6865

**Comment by graingert** [comment]
@rumpl that PR got merged, is there anything else blocking a release?

**Comment by graingert** [comment]
@rumpl @ulyssessouza that PR got merged, is there anything else blocking a release?

**Pull request merged**